Reduce doxygen warnings

johnnonweiler opened the pull request:
Following on from #3082, this addresses some more low hanging fruit for reducing the number of warnings when running doxygen.  Please see the individual commits for details.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

Commit:
Add missing newlines after doxygen \file tags

Commit:
Correct parameter names in doxygen comments

Commit:
Escape \ and # symbols in doxygen docs

Commit:
Change doxygen subsection label

Doxygen was giving a warning message about "multiple use of section
label 'deprecated'", so this commit changes the section label.  There
are no references to this section, so the label is only changed in
one place.  (I can't find any other uses of this section label, so
the warning message seems to be wrong, but it doesn't hurt to fix it.)

Commit:
Update params in doxygen comments

Delete documentation for parameters which no longer exist

chrisr-diffblue reviewed (approved):
The changes look basically sound, so I'm happy with those - but I note that CI is failing for the doxygen run, so that will need to be fixed before this is merged. I'm not immediately seeing why it is failing, so it may just be a travis glitch that is resolved with a build restart - but needs investigating.

tautschnig commented on src/util/README.md:
```diff
@@ -90,7 +90,7 @@ is initially populated from `irep_ids.def`, so for example the fourth entry
 You can refer to this \ref irep_idt as `ID_type`. The other kind of line you
 see is `“IREP_ID_TWO(C_source_location, #source_location)”`, which means the
-\ref irep_idt for the string “#source_location” can be referred to as
+\ref irep_idt for the string “\#source_location” can be referred to as
```
Maybe this change doesn't actually achieve what it is supposed to do? Doxygen now appears to be complaining that the link request could not be satisfied.

johnnonweiler replied:
This changed worked.  The problem is the other #source_location on the previous line.  (I'm not sure how I missed this earlier!)

This has turned out to be much more tricky to fix - adding a \ doesn't work in this case (as explained in https://stackoverflow.com/a/36750674 ).  I haven't found a perfect solution to this, but have now committed something that works with slightly different double quotes.

tautschnig replied:
I think this is good, especially with the elaborate commit message. It's not beautiful, but anyone can easily figure out what's going on here.

Commit:
Update comment to make doxygen happy

Doxygen produced a warning about a reference it couldn't find, because
of a '#' symbol used in the comment (which the author didn't intend to
be a reference).  Usually, '#' could be escaped '\#', but that doesn't
work in this case, because it appears within quotes (as explained in
https://stackoverflow.com/a/36750674 ).  I couldn't find a way to get
this to work with the quotes that had been used, so I have changed it
to use different quotes.

johnnonweiler commented:
@tautschnig, @chrisr-diffblue please could you let me know if you're still happy with this, given my additional commit?

chrisr-diffblue commented:
It's a bit ugly, but it does look like this is the only workaround to what seems like a weakness in Doxygen. As such, I'm OK with the additional tweak. I'm agnostic on whether this additional commit should be squashed into the earlier commit or not.

johnnonweiler commented:
I'd prefer not to squash it, as it is actually completely independent from the change in the previous commit.  (Travis is supposed to only complain about doxygen issues on lines which have been changed, but in this case doxygen was reporting the wrong line number in its warning message.)

allredj reviewed (commented):
This PR failed Diffblue compatibility checks (cbmc commit: eea5403d0a0f77b711bb1263dd937205b024a8ad).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87214638
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

allredj reviewed (commented):
Passed Diffblue compatibility checks (cbmc commit: eea5403d0a0f77b711bb1263dd937205b024a8ad).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87214638

NathanJPhillips reviewed (approved):
There are other mistakes on the lines you have fixed things on, but all your changes are an improvement so I approve.
